crates/pipeline_manager/src/compiler.rs

```diff
@@ -116,6 +122,69 @@ impl ProgramStatus {
     }
 }
 
+lazy_static::lazy_static! {
+    static ref METRICS: CompilerMetrics = init_metrics();
+}
+
+/// Compiler metrics to track the number of invocations and the latency, broken
+/// down by label, which is a pair of phase (SQL vs Rust) and status
+/// (success/error).
+pub struct CompilerMetrics {
+    invocations: Family<MetricLabel, Counter>,
+    latency: Family<MetricLabel, Histogram>,
+}
+
+/// We break down metrics in this file by the compiler phase and exit status
+/// These types need to implement EncodeLabelSet to allow the registry to render
+/// them for metrics scrape endpoints
+#[derive(Clone, Debug, Hash, PartialEq, Eq, EncodeLabelSet)]
+struct MetricLabel {
+    stage: StageType,
+    status: Status,
+}
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq, EncodeLabelValue)]
+enum StageType {
+    Sql,
+    Rust,
+}
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq, EncodeLabelValue)]
+enum Status {
+    Success,
+    Error,
+}
+
+fn init_metrics() -> CompilerMetrics {
+    CompilerMetrics {
+        invocations: Family::<MetricLabel, Counter>::default(),
+        latency: Family::<MetricLabel, Histogram>::new_with_constructor(|| {
+            // These are buckets for measuring SQL and rust compilation times
+            let buckets = [1.0, 5.0, 10.0, 20.0, 50.0, 100.0, 200.0, 400.0];
```

Comment:
what do the numbers mean?

Reply:
Histogram CDF points. So if compilation takes say, 15 seconds, it'll add 1 buckets to 20, 50, 100, 200 and 400.

I don't have a good sense for what the most useful CDF points should be yet, I hope to tweak them as we go.

Reply:
got it maybe this can be a comment :)